pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/impl/internal/NpathBaseVisitor.java

```diff
@@ -2,7 +2,7 @@
  * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
  */
 
-package net.sourceforge.pmd.lang.java.metrics.impl.visitors;
+package net.sourceforge.pmd.lang.java.metrics.impl.internal;
```

Comment:
is there any part of the `impl` package that is not going to be internal? should this simply be `net.sourceforge.pmd.lang.java.metrics.internal`?

Reply:
Well we could make metrics implementations internal too yes, but they're not deprecated at the moment. Shall I add the file movements in this PR and deprecated on 6.13.0 later?

Reply:
I'm ok with splitting it into stages, just wondered if `impl.internal` makes sense... is there a public implementation w want to keep?

Reply:
Not really, they're used as examples in the docs but they may be internal as well.

Reply:
On second thought, the abstract classes are in the `impl` package and they're published API. It would be more consistent to move them to the root package, beside `AbstractJavaMetric`